test/flycheck-test.el

```diff
@@ -3023,6 +3023,36 @@ of the file will be interrupted because there are too many #ifdef configurations
                                          :buffer 'buffer
                                          :filename "foo.php")))))
 
+(defconst flycheck-typescript-tslint-json
```

Comment:
Would you mind to port this to a buttercup test?  Take a look at `test/specs`, we already have a file for Haskell there, and you'd create a similar one for Typescript.  There's also an existing matcher already called `:to-be-equal-flycheck-errors` which compares to list of errors specifically, with much better error reporting than ert matchers.

We only use ERT for syntax checker integration tests, i.e. the other one you've written.